UIDS-GIDS.md

```diff
@@ -0,0 +1,190 @@
+# Users, Groups, UIDs and GIDs on `systemd` systems
+
+Here's a summary of the requirements `systemd` (and Linux) make on UID/GID
+assignments and their ranges.
+
+Note that while in theory UIDs and GIDs are orthogonal concepts they really
+aren't IRL. With that in mind, when we discuss UIDs below it should be assumed
+that whatever we say about about UIDs applies to GIDs the exact same way, and
+all the special assignments and ranges for UIDs always have the exact same
+validity for GIDs too. To say this clearly: systems that assign UIDs and GIDs
+of the same concepts independently of each other are not supported by `systemd`
+and void their warranty. That said, while users/UIDs generally have matching
```

Comment:
I don't thin we can say that. The way that things have always worked and still work is that when an admin once creates a group without creating a corresponding user, all user-group pairs created in the future will most likely have different numerical uids. That's just extremely common. [...] After reading the rest of patches, I don't even see why such a requirement would be added. Or maybe you mean something else? "assign concepts independently" is very unclear.